js/tabRestore.js

```diff
@@ -10,17 +10,35 @@ function TabHistory() {
 }
 
 TabHistory.prototype.push = function (tabId) {
-  var tab = tabs.get(tabId)
+  var closedTab = tabs.get(tabId)
+  var parentTask
+
+  tasks.get().forEach(function (task) {
+    task.tabs.forEach(function (tab) {
+      if (tab.id === tabId) {
+        parentTask = task
+      }
+    })
+  })
+
   // Do not store private tabs or blank tabs
-  if (tab.private || tab.url === 'about:blank' || tab.url === '') {
+  if (closedTab.private
+    || closedTab.url === 'about:blank'
+    || closedTab.url === '') {
     return
   }
 
   if (this.stack.length < this.depth) {
-    this.stack.push(tab)
+    this.stack.push({
+      tab: closedTab,
+      task: parentTask || getSelectedTask(),
```

Comment:
I don't remember what the reason was; it looks like it didn't even work when it was added in 71870a32d2c93b2eb5a63c56daba1bcc3137f119. I think I probably just added it while working on adding tasks support and then forgot about it. We can probably just get rid of it.